diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index c81b196..8935b87 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -211,8 +211,7 @@ if(${IMAGE_SUPPORT})
     HSA_IMAGE_SUPPORT
     UNIX_OS
     LINUX
-    __AMD64__
-    __x86_64__
+    __aarch64__
     AMD_INTERNAL_BUILD
     BRAHMA_BUILD=1 )
 
diff --git a/src/core/inc/amd_memory_region.h b/src/core/inc/amd_memory_region.h
index cce01e6..8a39a73 100644
--- a/src/core/inc/amd_memory_region.h
+++ b/src/core/inc/amd_memory_region.h
@@ -187,7 +187,7 @@ class MemoryRegion : public core::MemoryRegion {
 
   mutable KernelMutex access_lock_;
 
-  static const size_t kPageSize_ = 4096;
+  static size_t kPageSize_;
 
   // Determine access type allowed to requesting device
   hsa_amd_memory_pool_access_t GetAccessInfo(const core::Agent& agent,
diff --git a/src/core/runtime/amd_memory_region.cpp b/src/core/runtime/amd_memory_region.cpp
index 4419286..63a5079 100644
--- a/src/core/runtime/amd_memory_region.cpp
+++ b/src/core/runtime/amd_memory_region.cpp
@@ -50,12 +50,14 @@
 #include "core/inc/amd_gpu_agent.h"
 #include "core/util/utils.h"
 #include "core/inc/exceptions.h"
+#include <unistd.h>
 
 namespace rocr {
 namespace AMD {
 
 // Tracks aggregate size of system memory available on platform
 size_t MemoryRegion::max_sysmem_alloc_size_ = 0;
+size_t MemoryRegion::kPageSize_ = sysconf(_SC_PAGESIZE);
 
 void* MemoryRegion::AllocateKfdMemory(const HsaMemFlags& flag,
                                       HSAuint32 node_id, size_t size) {
@@ -123,7 +125,7 @@ MemoryRegion::MemoryRegion(bool fine_grain, bool kernarg, bool full_profile, cor
 
     virtual_size_ = kGpuVmSize;
   } else if (IsSystem()) {
-    mem_flag_.ui32.PageSize = HSA_PAGE_SIZE_4KB;
+    mem_flag_.ui32.PageSize = MemoryRegion::kPageSize_;
     mem_flag_.ui32.NoSubstitute = 0;
     mem_flag_.ui32.HostAccess = 1;
     mem_flag_.ui32.CachePolicy = HSA_CACHING_CACHED;
diff --git a/src/image/util.h b/src/image/util.h
index 5eb9f3a..9f03d98 100644
--- a/src/image/util.h
+++ b/src/image/util.h
@@ -74,6 +74,8 @@ namespace image {
 #include "mm_malloc.h"
 #if defined(__i386__) || defined(__x86_64__)
 #include <x86intrin.h>
+#elif defined(__aarch64__)
+
 #else
 #error                                                                                             \
     "Processor not identified.  " \
diff --git a/src/inc/hsa.h b/src/inc/hsa.h
index fe1facc..3a58546 100644
--- a/src/inc/hsa.h
+++ b/src/inc/hsa.h
@@ -80,7 +80,7 @@
 // Try to detect CPU endianness
 #if !defined(LITTLEENDIAN_CPU) && !defined(BIGENDIAN_CPU)
 #if defined(__i386__) || defined(__x86_64__) || defined(_M_IX86) || \
-    defined(_M_X64)
+    defined(_M_X64) || defined(__aarch64__)
 #define LITTLEENDIAN_CPU
 #endif
 #endif
