diff --git a/CMakeLists.txt b/CMakeLists.txt
index d102066..67d2343 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -61,7 +61,7 @@ endif()
 
 ## Compiler flags
 set(CMAKE_CXX_FLAGS
- "${CMAKE_CXX_FLAGS} -Wall -Wextra -fno-rtti -m64 -msse -msse2 -std=c++11 ")
+ "${CMAKE_CXX_FLAGS} -Wall -Wextra -fno-rtti -std=c++11 ")
 # Security options
 set(CMAKE_CXX_FLAGS
  "${CMAKE_CXX_FLAGS} -Wconversion -Wcast-align ")
diff --git a/tests/rocm_smi_test/CMakeLists.txt b/tests/rocm_smi_test/CMakeLists.txt
index b517184..cd95496 100755
--- a/tests/rocm_smi_test/CMakeLists.txt
+++ b/tests/rocm_smi_test/CMakeLists.txt
@@ -156,11 +156,6 @@ endif()
 #
 # Extend the compiler flags for 64-bit builds
 #
-if (IS64BIT)
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -m64  -msse -msse2")
-else()
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -m32")
-endif()
 
 #
 # Add compiler flags to include symbol information for debug builds
diff --git a/tests/rocm_smi_test/functional/power_read.cc b/tests/rocm_smi_test/functional/power_read.cc
index 70f3b10..a8e5f8b 100755
--- a/tests/rocm_smi_test/functional/power_read.cc
+++ b/tests/rocm_smi_test/functional/power_read.cc
@@ -104,11 +104,13 @@ void TestPowerRead::Run(void) {
       IF_VERB(STANDARD) {
         std::cout << "\t**Current Power Cap: " << val_ui64 << "uW" <<std::endl;
       }
-      err = rsmi_dev_power_cap_default_get(i, &val_ui64);
+
+      //kfd driver in kernel 5.10 is not support Default Power Cap
+/*      err = rsmi_dev_power_cap_default_get(i, &val_ui64);
       CHK_ERR_ASRT(err)
       IF_VERB(STANDARD) {
         std::cout << "\t**Default Power Cap: " << val_ui64 << "uW" <<std::endl;
-      }
+      } */
       err = rsmi_dev_power_cap_range_get(i, 0, &val_ui64, &val2_ui64);
       CHK_ERR_ASRT(err)
       IF_VERB(STANDARD) {
